### PR TITLE
Stop scabbard upon shutdown signal

### PR DESCRIPTION
**Pull request by eloaverona**
This PR add a method to AdminServiceShared to stop all running services, and adds a call to that method up admin service shutdown.

### CODE REVIEW

File: libsplinter/src/admin/shared.rs [hunk 1]
```diff
@@ -1070,6 +1070,30 @@ impl AdminServiceShared {
...
     }
 
+    /// Stops all running services
+    pub fn stop_services(&mut self) -> Result<(), AdminSharedError> {
+        let running_services = self
+            .running_services
+            .drain()
+            .collect::<Vec<ServiceDefinition>>();
+        let shutdown_errors = running_services
+            .into_iter()
```
Comment (ltseeley) [inline libsplinter/src/admin/shared.rs]: Looks like this is converting an iterator -> vec -> back to iterator.

File: libsplinter/src/admin/shared.rs [hunk 2]
```diff
@@ -1070,6 +1070,30 @@ impl AdminServiceShared {
...
+            .map(|service| {
+                debug!(
+                    "Stoping service {} in circuit {}",
+                    service.service_type, service.circuit
+                );
+                self.orchestrator.shutdown_service(&service)
+            })
+            .filter_map(|res| if let Err(err) = res { Some(err) } else { None })
```
Comment (ltseeley) [inline libsplinter/src/admin/shared.rs]: Could probably do `.filter_map(Result::err)`

File: libsplinter/src/admin/mod.rs [hunk 3]
```diff
@@ -188,6 +188,10 @@ impl Service for AdminService {
         // Shutdown event dealers and disconnect active websocket connections
         admin_service_shared.shutdown_event_dealers();
 
+        admin_service_shared
+            .stop_services()
```
Comment (ltseeley) [inline libsplinter/src/admin/mod.rs]: Typo in commit message: up -> upon

File: libsplinter/src/service/scabbard/mod.rs [hunk 4]
```diff
@@ -201,6 +201,7 @@ impl Service for Scabbard {
         &mut self,
         service_registry: &dyn ServiceNetworkRegistry,
     ) -> Result<(), ServiceStopError> {
+        debug!("Stopping scabbard");
```
Comment (ltseeley) [inline libsplinter/src/service/scabbard/mod.rs]: I'd include the service ID here (for when there's more than once scabbard service running).

File: libsplinter/src/admin/shared.rs (hunk 2 above)
Comment (eloaverona) [reply libsplinter/src/admin/shared.rs]: updated

File: libsplinter/src/admin/mod.rs (hunk 3 above)
Comment (eloaverona) [reply libsplinter/src/admin/mod.rs]: updated

File: libsplinter/src/service/scabbard/mod.rs (hunk 4 above)
Comment (eloaverona) [reply libsplinter/src/service/scabbard/mod.rs]: updated

**Pull request merged**